src/proposals/20250403/SparkEthereum_20250403.t.sol

```diff
@@ -0,0 +1,457 @@
+        vm.prank(CENTRIFUGE_ROOT);
+        IInvestmentManager(CENTRIFUGE_INVESTMENT_MANAGER).fulfillRedeemRequest(
+            CENTRIFUGE_JTRSY_POOL_ID,
+            CENTRIFUGE_JTRSY_TRANCHE_ID,
+            address(ctx.proxy),
+            CENTRIFUGE_USDC_ASSET_ID,
+            _amountJtrsy * 2,
+            _amountJtrsy
+        );
+    }
+
+    function test_ETHEREUM_mapleSyrupUSDCOnboarding() public onChain(ChainIdUtils.Ethereum()) {
+        address vault                 = SYRUP_USDC;
+        uint256 expectedDepositAmount = 25_000_000e6;
+        uint256 depositMax            = 25_000_000e6;
+        uint256 depositSlope          = 5_000_000e6 / uint256(1 days);
+        IERC20 usdc                   = IERC20(Ethereum.USDC);
+        IMapleTokenExtended syrup     = IMapleTokenExtended(vault);
+        
+        // Slightly modify code from _testERC4626Onboarding due to async redemption
+        SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
+        bool unlimitedDeposit = depositMax == type(uint256).max;
+        bytes32 depositKey = RateLimitHelpers.makeAssetKey(
+            MainnetController(ctx.controller).LIMIT_4626_DEPOSIT(),
+            vault
+        );
+        bytes32 withdrawKey = RateLimitHelpers.makeAssetKey(
+            MainnetController(ctx.controller).LIMIT_4626_WITHDRAW(),
+            vault
+        );
+
+        _assertRateLimit(depositKey, 0, 0);
+        _assertRateLimit(withdrawKey, 0, 0);
+
+        vm.prank(ctx.relayer);
+        vm.expectRevert("RateLimits/zero-maxAmount");
+        MainnetController(ctx.controller).depositERC4626(vault, expectedDepositAmount);
+
+        executeAllPayloadsAndBridges();
+
+        ctx.controller = ETHEREUM_NEW_ALM_CONTROLLER;
+
+        vm.startPrank(ctx.relayer);
+        MainnetController(ctx.controller).mintUSDS(expectedDepositAmount * 1e12);
+        MainnetController(ctx.controller).swapUSDSToUSDC(expectedDepositAmount);
+        vm.stopPrank();
+
+        _assertRateLimit(depositKey, depositMax, depositSlope);
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        if (!unlimitedDeposit) {
+            vm.prank(ctx.relayer);
+            vm.expectRevert("RateLimits/rate-limit-exceeded");
+            MainnetController(ctx.controller).depositERC4626(vault, depositMax + 1);
+        }
+
+        assertEq(ctx.rateLimits.getCurrentRateLimit(depositKey),  depositMax);
+        assertEq(ctx.rateLimits.getCurrentRateLimit(withdrawKey), type(uint256).max);
+        assertEq(usdc.balanceOf(address(ctx.proxy)),              expectedDepositAmount);
+        assertEq(syrup.balanceOf(address(ctx.proxy)),             0);
+
+        vm.prank(ctx.relayer);
+        uint256 shares = MainnetController(ctx.controller).depositERC4626(vault, expectedDepositAmount);
+        assertGt(shares, 0);
+
+        assertEq(ctx.rateLimits.getCurrentRateLimit(depositKey),  unlimitedDeposit ? type(uint256).max : depositMax - expectedDepositAmount);
```

Comment:
Can just be zero cause whole 25m was consumed

Reply:
changed